nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/flow-analysis-rules/flow-analysis-rules.effects.ts

```diff
@@ -0,0 +1,441 @@
+        () =>
+            this.actions$.pipe(
+                ofType(FlowAnalysisRuleActions.navigateToEditFlowAnalysisRule),
+                map((action) => action.id),
+                tap((id) => {
+                    this.router.navigate(['/settings', 'flow-analysis-rules', id, 'edit']);
+                })
+            ),
+        { dispatch: false }
+    );
+
+    openConfigureFlowAnalysisRuleDialog$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(FlowAnalysisRuleActions.openConfigureFlowAnalysisRuleDialog),
+                map((action) => action.request),
+                tap((request) => {
+                    const taskId: string = request.id;
+
+                    const editDialogReference = this.dialog.open(EditFlowAnalysisRule, {
+                        data: {
+                            flowAnalysisRule: request.flowAnalysisRule
+                        },
+                        id: taskId,
+                        panelClass: 'large-dialog'
+                    });
+
+                    editDialogReference.componentInstance.saving$ = this.store.select(selectSaving);
+
+                    editDialogReference.componentInstance.createNewProperty = (
+                        existingProperties: string[],
+                        allowsSensitive: boolean
+                    ): Observable<Property> => {
+                        const dialogRequest: NewPropertyDialogRequest = { existingProperties, allowsSensitive };
+                        const newPropertyDialogReference = this.dialog.open(NewPropertyDialog, {
+                            data: dialogRequest,
+                            panelClass: 'small-dialog'
+                        });
+
+                        return newPropertyDialogReference.componentInstance.newProperty.pipe(
+                            take(1),
+                            switchMap((dialogResponse: NewPropertyDialogResponse) => {
+                                return this.flowAnalysisRuleService
+                                    .getPropertyDescriptor(request.id, dialogResponse.name, dialogResponse.sensitive)
+                                    .pipe(
+                                        take(1),
+                                        map((response) => {
+                                            newPropertyDialogReference.close();
+
+                                            return {
+                                                property: dialogResponse.name,
+                                                value: null,
+                                                descriptor: response.propertyDescriptor
+                                            };
+                                        })
+                                    );
+                            })
+                        );
+                    };
+
+                    const goTo = (commands: string[], destination: string): void => {
```

Comment:
This is currently unused. I think go to Controller Service is missing.

Reply:
It seems it is also is unused in current nifi but since it was there I added it here...